source/includes/fundamentals/code-snippets/vectorSearchQuery.go

```diff
@@ -0,0 +1,85 @@
+
+	// Converts the BSON vector to a BSON binary vector
+	queryVector := vectorFloat.Binary()
+	// end-binary-vector
+
+	// start-convert-back-vector
+	// Converts the BSON binary vector back to a BSON vector
+	backToVectorFloat, err := bson.NewVectorFromBinary(queryVector)
+	if err != nil {
+		log.Fatal(err)
+	}
+	// end-convert-back-vector
+
+	// start-aggregation
+	// Creates the aggregation pipeline
+	vectorSearchStage := bson.D{
+		{"$vectorSearch", bson.D{
+			{"index", "vector_index"},
+			{"path", "plot_embedding"},
+			{"queryVector", queryVector},
+			{"numCandidates", 150},
+			{"limit", 5},
+		}}}
+
+	projectStage := bson.D{
+		{"$project", bson.D{
+			{"_id", 0},
+			{"plot", 1},
+			{"title", 1},
+		}}}
+
+	// Runs the aggregation pipeline
+	cursor, err := coll.Aggregate(ctx, mongo.Pipeline{vectorSearchStage, projectStage})
+	if err != nil {
+		log.Fatalf("failed to retrieve data from the server: %v", err)
+	}
+
+	// Displays the results
+	type ProjectedMovieResult struct {
+		Title string `bson:"title"`
+		Plot  string `bson:"plot"`
+	}
+
+	var results []ProjectedMovieResult
+	if err = cursor.All(ctx, &results); err != nil {
+		log.Fatalf("failed to unmarshal retrieved docs to ProjectedMovieResult objects: %v", err)
+	}
+
+	for _, result := range results {
+		fmt.Printf("Title: %v \nPlot: %v \nScore: %v \n\n", result.Title, result.Plot, result.Score)
```

Comment:
We need to add `Score` in the `ProjectedMovieResult` struct.

Reply:
Done

Reply:
I apologize for the back and forth.

Do we also need to update:
https://github.com/mongodb/docs-golang/blob/3af27fd57484c19d02944807400700abe99a684d/source/fundamentals/atlas-vector-search.txt#L117
Otherwise, we can just remove the `Score` in the `fmt.Printf()`.